torch/optim/rmsprop.py

```diff
@@ -49,7 +56,15 @@ def step(self, closure=None):
                     grad = grad.add(group['weight_decay'], p.data)
 
                 square_avg.mul_(alpha).addcmul_(1 - alpha, grad, grad)
-                avg = square_avg.sqrt().add_(group['eps'])
-                p.data.addcdiv_(-group['lr'], grad, avg)
+
+                avg = None
+                if group['centered']:
+                    grad_avg.mul_(alpha).add_(1 - alpha, grad)
+                    avg = square_avg.addcmul(-1, grad_avg, grad_avg).sqrt().add_(group['eps'])
+                else:
+                    avg = square_avg.sqrt().add_(group['eps'])
+
+                momentum.mul_(group['momentum']).addcdiv_(group['lr'], grad, avg)
```

Comment:
I think it's best to guard this line and the allocation of momentum buffer with an additional `if`. It will save some multiplications, but I think more importantly it will decrease the mem usage by the size of parameters.